editor.planx.uk/src/@planx/components/List/Public/Fields.tsx

```diff
@@ -0,0 +1,137 @@
+import Box from "@mui/material/Box";
+import FormControl from "@mui/material/FormControl";
+import FormLabel from "@mui/material/FormLabel";
+import MenuItem from "@mui/material/MenuItem";
+import RadioGroup from "@mui/material/RadioGroup";
+import React from "react";
+import SelectInput from "ui/editor/SelectInput";
+import InputLabel from "ui/public/InputLabel";
+import Input from "ui/shared/Input";
+import InputRowLabel from "ui/shared/InputRowLabel";
+
+import { DESCRIPTION_TEXT } from "../../shared/constants";
+import BasicRadio from "../../shared/Radio/BasicRadio";
+import type { NumberField, QuestionField, TextField } from "../model";
+
+type Props<T> = T & { id: string };
+
+export const TextFieldInput: React.FC<Props<TextField>> = ({
+  id,
+  data,
+  required,
+}) => (
+  <InputLabel label={data.title} htmlFor={id}>
+    <Input
+      type={((type) => {
+        if (type === "email") return "email";
+        else if (type === "phone") return "tel";
+        return "text";
+      })(data.type)}
+      multiline={data.type && ["long", "extraLong"].includes(data.type)}
```

Comment:
There's some repeated code here from existing components. 

I'm hoping that once error handling / form values are handled I can DRY this up a little with more generic, shared, input components.